File: abc/065/c.rb
D = 1000000007

n, m = gets.split.map(&:to_i)
diff = (n - m).abs

def fact(n)
  return 1 if n == 0

  1.upto(n).reduce(1, &:*)
end

if diff == 0
  puts 2 * fact(n) * fact(m) % D
elsif diff == 1
  puts fact(n) * fact(m) % D
else
  puts 0
end

File: abc/065/c.py
import math

divider = 1000000007

n, m = map(int, input().split())
diff = abs(n - m)

if diff == 0:
    print(2 * math.factorial(n) * math.factorial(m) % divider)
elif diff == 1:
    print(math.factorial(n) * math.factorial(m) % divider)
else:
    print(0)
